x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -265,6 +266,54 @@ export function registerIndexRoutes({
     })
   );
 
+  router.post(
```

Comment:
suggestion: worth moving the ML routes into a separate route file. I can imagine this is going to get large pretty quickly.

Reply:
I'll add a tech debt ticket for this in order to keep the scope of this PR small.

Reply:
Just a note - you'll also want to add a test for the route as well. With this [backend PR merged yesterday](https://github.com/elastic/kibana/pull/140459), I [started a test file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.test.ts) for the routes under `indices` where we can add these.

(as mentioned, we will add a new issue as well as a stretch goal to move the ML-specific routes out of here and to their own file - we can probably do this after FF)

Reply:
Filed Filed https://github.com/elastic/enterprise-search-team/issues/2752 for the refactoring. Also added route test.